File: app/src/main/java/com/example/sunnygym/ApiService.java
package com.example.sunnygym;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import java.util.List;

public interface ApiService {
    @GET("reservas")
    Call<List<Reserva>> getReservas();

    @POST("reservas")
    Call<Reserva> addReserva(@Body Reserva reserva);

    @DELETE("reservas/{id}")
    Call<Void> deleteReserva(@Path("id") int id);

    @PUT("reservas/{id}")
    Call<Reserva> updateReserva(@Path("id") int id, @Body Reserva reserva);

    @GET("clases_disponibles")
    Call<List<String>> getClasesDisponibles();


}

File: APP_Kotlin/app/src/main/java/com/example/sunnygym/ApiService.kt
package com.example.sunnygym

import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    @GET("reservas")
    suspend fun getReservas(): Response<List<Reserva>>

    @POST("reservas")
    suspend fun addReserva(@Body reserva: Reserva): Response<Reserva>

    @DELETE("reservas/{id}")
    suspend fun deleteReserva(@Path("id") id: Int): Response<Void>

    @PUT("reservas/{id}")
    suspend fun updateReserva(@Path("id") id: Int, @Body reserva: Reserva): Response<Reserva>

    @GET("clases_disponibles")
    suspend fun getClasesDisponibles(): Response<List<String>>
}